tests/components/foscam/test_init.py

```diff
@@ -0,0 +1,137 @@
+"""Test the Foscam component."""
+
+from unittest.mock import patch
+
+from homeassistant.components.foscam import DOMAIN, config_flow
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from .conftest import setup_mock_foscam_camera
+from .const import ENTRY_ID, VALID_CONFIG
+
+from tests.common import MockConfigEntry
+
+
+async def test_unique_id_new_entry(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test unique ID for a newly added device is correct."""
+    entry = MockConfigEntry(
+        domain=config_flow.DOMAIN,
+        data=VALID_CONFIG,
+        entry_id=ENTRY_ID,
+    )
+    entry.add_to_hass(hass)
+
+    with (
+        # We are not interested in the entry migration here
+        patch(
+            "homeassistant.components.foscam.async_migrate_entry",
+            return_value=(True),
+        ),
+        patch("homeassistant.components.foscam.FoscamCamera") as mock_foscam_camera,
+        patch(
+            "homeassistant.components.foscam.camera.HassFoscamCamera.async_added_to_hass",
+        ),
```

Comment:
Why do we patch out the migration and the async_added_to_hass? Ideally we patch no internals

Reply:
The trick for the migration not to run is to set the version of the config entry to 2, but only if the entry is what a version 2 config entry looks like